// @dicebear/avatars/lib/model/color/collection/modifier/lightness/difference.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var brighter_1 = require("./brighter");
var darker_1 = require("./darker");
/**
 * Ensures that the selected color is darker or brighter than a reference color.
 */
var Difference = /** @class */ (function () {
    /**
     * @param colors
     * @param reference
     * @param difference
     */
    function Difference(primaryCollection, secondaryCollection, difference) {
        this.pickedColors = {};
        this.primaryCollection = primaryCollection;
        this.secondaryCollection = secondaryCollection;
        this.brighter = new brighter_1.default(primaryCollection, secondaryCollection, difference);
        this.darker = new darker_1.default(primaryCollection, secondaryCollection, difference);
    }
    /**
     * Returns a color
     *
     * @param random
     */
    Difference.prototype.get = function (random) {
        if (this.pickedColors[random.seed]) {
            return this.pickedColors[random.seed];
        }
        var primaryColor = this.primaryCollection.get(random);
        var secondaryColor = this.secondaryCollection.get(random);
        var primaryColorHsv = primaryColor.hsv;
        var secondaryColorHsv = secondaryColor.hsv;
        if (primaryColorHsv[2] <= secondaryColorHsv[2]) {
            return (this.pickedColors[random.seed] = this.darker.get(random));
        }
        else {
            return (this.pickedColors[random.seed] = this.brighter.get(random));
        }
    };
    return Difference;
}());
exports.default = Difference;
